File: frontend/src/services/filial.service.ts
import { api } from './api'

export interface FilialResponse {
  id: string; codigo: string; nome: string; razaoSocial: string; cnpj: string
  inscricaoEstadual?: string; endereco?: string; numero?: string; cidade: string
  uf: string; cep?: string; telefone?: string; email?: string
  horarioInicio?: string; horarioFim?: string; limiteDiario?: number
  tempoMedioAtendimento?: number; ativo: boolean; createdAt: string
}

export interface FilialRequest {
  codigo: string; nome: string; razaoSocial: string; cnpj: string
  inscricaoEstadual?: string; endereco?: string; numero?: string
  complemento?: string; bairro?: string; cidade: string; uf: string
  cep?: string; telefone?: string; email?: string
  horarioInicio?: string; horarioFim?: string
  limiteDiario?: number; tempoMedioAtendimento?: number; ativo?: boolean
}

export interface PageResponse<T> {
  content: T[]; totalElements: number; totalPages: number; size: number; number: number
}

export const filialService = {
  listar: async (busca?: string, page = 0, size = 20): Promise<PageResponse<FilialResponse>> => {
    const params = new URLSearchParams({ page: String(page), size: String(size) })
    if (busca) params.set('busca', busca)
    const { data } = await api.get(`/api/v1/filiais?${params}`)
    return data.data
  },

  buscarPorId: async (id: string): Promise<FilialResponse> => {
    const { data } = await api.get(`/api/v1/filiais/${id}`)
    return data.data
  },

  criar: async (request: FilialRequest): Promise<FilialResponse> => {
    const { data } = await api.post('/api/v1/filiais', request)
    return data.data
  },

  atualizar: async (id: string, request: FilialRequest): Promise<FilialResponse> => {
    const { data } = await api.put(`/api/v1/filiais/${id}`, request)
    return data.data
  },

  excluir: async (id: string): Promise<void> => {
    await api.delete(`/api/v1/filiais/${id}`)
  },
}

File: backend/src/main/java/com/rbm/agendamento/application/service/FilialService.java
package com.rbm.agendamento.application.service;

import com.rbm.agendamento.api.exception.BusinessException;
import com.rbm.agendamento.api.exception.ResourceNotFoundException;
import com.rbm.agendamento.application.dto.filial.FilialRequest;
import com.rbm.agendamento.application.dto.filial.FilialResponse;
import com.rbm.agendamento.domain.entity.Filial;
import com.rbm.agendamento.infrastructure.repository.FilialRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.UUID;

@Service
@RequiredArgsConstructor
public class FilialService {

    private final FilialRepository filialRepository;

    @Transactional(readOnly = true)
    public Page<FilialResponse> listar(String busca, Pageable pageable) {
        return filialRepository.buscar(busca, pageable).map(FilialResponse::from);
    }

    @Transactional(readOnly = true)
    public List<FilialResponse> listarAtivas() {
        return filialRepository.findAllByAtivoTrue().stream().map(FilialResponse::from).toList();
    }

    @Transactional(readOnly = true)
    public FilialResponse buscarPorId(UUID id) {
        return FilialResponse.from(buscarEntidade(id));
    }

    @Transactional
    public FilialResponse criar(FilialRequest request) {
        if (filialRepository.existsByCodigo(request.codigo()))
            throw new BusinessException("CODIGO_DUPLICADO", "Código de filial já cadastrado: " + request.codigo());
        if (filialRepository.existsByCnpj(request.cnpj()))
            throw new BusinessException("CNPJ_DUPLICADO", "CNPJ já cadastrado: " + request.cnpj());

        Filial filial = mapear(new Filial(), request);
        return FilialResponse.from(filialRepository.save(filial));
    }

    @Transactional
    public FilialResponse atualizar(UUID id, FilialRequest request) {
        Filial filial = buscarEntidade(id);

        if (!filial.getCodigo().equals(request.codigo()) && filialRepository.existsByCodigo(request.codigo()))
            throw new BusinessException("CODIGO_DUPLICADO", "Código de filial já cadastrado: " + request.codigo());
        if (!filial.getCnpj().equals(request.cnpj()) && filialRepository.existsByCnpj(request.cnpj()))
            throw new BusinessException("CNPJ_DUPLICADO", "CNPJ já cadastrado: " + request.cnpj());

        return FilialResponse.from(filialRepository.save(mapear(filial, request)));
    }

    @Transactional
    public void excluir(UUID id) {
        Filial filial = buscarEntidade(id);
        filial.softDelete();
        filialRepository.save(filial);
    }

    private Filial buscarEntidade(UUID id) {
        return filialRepository.findById(id)
                .orElseThrow(() -> ResourceNotFoundException.of("Filial", id));
    }

    private Filial mapear(Filial filial, FilialRequest r) {
        filial.setCodigo(r.codigo());
        filial.setNome(r.nome());
        filial.setRazaoSocial(r.razaoSocial());
        filial.setCnpj(r.cnpj());
        filial.setInscricaoEstadual(r.inscricaoEstadual());
        filial.setEndereco(r.endereco());
        filial.setNumero(r.numero());
        filial.setComplemento(r.complemento());
        filial.setBairro(r.bairro());
        filial.setCidade(r.cidade());
        filial.setUf(r.uf());
        filial.setCep(r.cep());
        filial.setLatitude(r.latitude());
        filial.setLongitude(r.longitude());
        filial.setTelefone(r.telefone());
        filial.setEmail(r.email());
        filial.setHorarioInicio(r.horarioInicio());
        filial.setHorarioFim(r.horarioFim());
        filial.setLimiteDiario(r.limiteDiario());
        filial.setTempoMedioAtendimento(r.tempoMedioAtendimento());
        if (r.ativo() != null) filial.setAtivo(r.ativo());
        return filial;
    }
}
